grub-core/loader/multiboot_mbi2.c

```diff
@@ -390,6 +398,34 @@ grub_multiboot2_load (grub_file_t file, const char *filename)
     err = grub_multiboot2_set_console (GRUB_MULTIBOOT2_CONSOLE_EGA_TEXT,
 				       accepted_consoles,
 				       0, 0, 0, console_required);
+
+  if (grub_slaunch_platform_type () == SLP_INTEL_TXT)
```

Comment:
Together with change in skinit.c:

```suggestion
  if (grub_slaunch_platform_type () != SLP_NONE)
```

Reply:
https://github.com/TrenchBoot/grub/compare/58f381041fe9f418529a53abb339ce78f1d38924..ccf8d46f1529cbc011ecd0b0ad02957ef7e40c4b#diff-285279252527f5bc911013189ac09a0880271f60dc8f28f28fe68c2f5d517c74R402